nx/lib/nx.ex

```diff
@@ -1212,18 +1271,25 @@ defmodule Nx do
         backend.iota(%T{type: type, shape: shape, names: names}, nil, backend_options)
       end
 
-    if not is_nil(vectorized_axes) and vectorized_axes != [] do
-      base_shape =
-        List.to_tuple(List.duplicate(1, length(vectorized_axes)) ++ Tuple.to_list(shape))
+    result =
+      if not is_nil(vectorized_axes) and vectorized_axes != [] do
+        base_shape =
+          List.to_tuple(List.duplicate(1, length(vectorized_axes)) ++ Tuple.to_list(shape))
 
-      output_shape = List.to_tuple(Keyword.values(vectorized_axes) ++ Tuple.to_list(shape))
+        output_shape = List.to_tuple(Keyword.values(vectorized_axes) ++ Tuple.to_list(shape))
 
-      output
-      |> reshape(base_shape)
-      |> broadcast(output_shape)
-      |> vectorize(vectorized_axes)
+        output
+        |> reshape(base_shape)
+        |> broadcast(output_shape)
+        |> vectorize(vectorized_axes)
+      else
+        output
+      end
+
+    if strides do
+      window_min(result, Tuple.duplicate(1, length(strides)), strides: strides)
     else
-      output
+      result
```

Comment:
This will probably make us hold on to the iota for longer than needed in EXLA or other backends. I implemented it here first for validation on the concept and interface, but perhaps we should do it on each backend.

libtorch arange does provide this, but XLA iota does not (jax does it by building the strided iota from np.arange instead).

Thoughts?